### DISCUSSION TITLE
Help getting notification on iOS

### DISCUSSION THREAD

**Discussion by vaibhav-sukhdeve** [thomasgalliker/Plugin.FirebasePushNotifications#93] [Q&A]
Hi,
First of all thanks for the Nuget package. 
It is working fine for android. But when I am trying to get the notification on iOS 18 I am unable to get the notification. Will you please help me find what I am doing wrong?

I am able to generate the token but when I send notification using the same token I do not get notification

**Comment by thomasgalliker** [comment]
- What is the content of your notification message? Something like this?
```
{
    "message": {
        "token": "{{fcm_token}}",
        "notification": {
            "title": "Notification title",
            "body": "Notification body"
        }
    }
}
```
- Did you ask for notification permissions?
- Did you subscribe to the IFirebasePushNotification.NotificationReceived event?
- Did you try the sample app provided in this repository? You can replace the Firebase google service files (Android/iOS) with your own files and send notification to this sample app.

**Reply by vaibhav-sukhdeve** [reply]
- The content of Notification Message is like this
`{
    "message": {
        "token": "fKcn9xv-RG28nMYLeco1BL:APA91bHPqgMFIvzRRnSDfBmSl_0eC7LF_6z-uITv16RVndfoM31_TixbQwLEtV9JpwEBN20b-QvZNn3ie_CZtOj1Q9k6-JjHKwn8tl2UyYm5MkF7-ADcKiydXiAEqSMoulW3J-wqb5wJ",
        "data": {
            "body": "Body of Your Notification in data",
            "title": "Title of Your Notification in data",
            "key_1": "Value for key_1",
            "key_2": "Value for key_2"
        }
    }
}`

- I am getting the Notification Permission Request
- Yes, I subscribed to the event and I am getting the notification on Android

**Comment by thomasgalliker** [comment]
- Did you add the P8 file in Firebase console?
- Did you place the GoogleService-Info.plist in the right place and reference it the same way as I did in the sample app's csproj file?

**Reply by vaibhav-sukhdeve** [reply]
- Yes P8 file is added 
- This is how I defined the file 
                  <ItemGroup Condition="$(TargetFramework.Contains('-ios'))">
			<BundleResource Include="Platforms\iOS\Resources\GoogleService-Info.plist" Link="GoogleService-Info.plist" />
		</ItemGroup>

I am wondering what is missing. 
In sample code may I know where is the code for displaying the notification.

**Comment by seanjaeger73** [comment]
I am going to piggyback off this post since we are running into a similar issue.  We migrated off Shiny.NET to this package mainly for the simplicity and I was told they will no longer be supporting Firebase.

I am running into the same issue and as @vaibhav-sukhdeve stated, all the above is correct in our code.  We changed very little from Shiny.NET to your solution as far as the files/how they are referenced and such (and Firebase settings)... but iOS notifications were working with Shiny, and not with this solution.  Android is working great!

When we start the iOS app we get the permission request.  I get a token, and when looking at the settings on iOS the permissions are set correctly.  Using the firebase test tool, I send a message and only receive on Android.  And all the events gets fired correctly with Android as well...

Anything else that I might be missing?  Thank you in advance... appreciate all your hard work on this.
